web/packages/selfhosted/test/utils.ts

```diff
@@ -117,30 +116,66 @@ export async function setupAndPlay(
 export async function getTraceOutput(
     browser: WebdriverIO.Browser,
     player: ChainablePromiseElement,
-) {
-    // Await any trace output
+    messageCount: number = 1,
+): Promise<string[]> {
+    // Await trace output
     await browser.waitUntil(
         async () => {
-            return (
-                (await browser.execute(
-                    (player) => {
-                        return (player as Player.PlayerElement).__ruffle_log__;
-                    },
-                    await player,
-                )) !== ""
+            const log = await browser.execute(
+                (player) => {
+                    return (player as Player.PlayerElement).__ruffle_log__;
+                },
+                await player,
             );
+            return log.length >= messageCount;
         },
         {
-            timeoutMsg: "Expected Ruffle to trace a message",
+            timeoutMsg: `Expected Ruffle to trace ${messageCount} messages`,
+        },
+    );
+
+    // Get the output
+    return await browser.execute(
+        (playerElement, messageCount) => {
+            const player = playerElement as Player.PlayerElement;
+            const m = messageCount as unknown as number;
+            return player.__ruffle_log__.splice(0, m);
         },
+        player,
+        messageCount,
     );
+}
+
+export async function expectTraceOutput(
+    browser: WebdriverIO.Browser,
+    player: ChainablePromiseElement,
+    messages: string[],
+) {
+    expect(
+        await getTraceOutput(browser, player, messages.length),
+    ).to.deep.equal(messages);
+}
+
+export async function clearTraceOutput(
+    browser: WebdriverIO.Browser,
+    player: ChainablePromiseElement,
+) {
+    await browser.execute((playerElement) => {
+        const player = playerElement as Player.PlayerElement;
+        player.__ruffle_log__ = [];
+    }, player);
+}
```

Comment:
For debugging the flakiness: Is it possible that the initial call to this is sometimes too late, and drops valuable output? :thinking: (I may be way off here, apologies in that case.)